pkg/controller/operators/adoption_controller.go

```diff
@@ -235,6 +237,14 @@ func (r *AdoptionReconciler) adoptComponents(ctx context.Context, csv *operators
 			}()
 		}
 
+		// alert on components that are deprecated in upcoming kubernetes versions
```

Comment:
Do we need to emit this on every reconciliation? Are we potentially going to start spewing metrics far too frequently in some cases? Can we batch these somehow?

I'm wondering if we should be aggregating this kind of metric and have a separate process that emits them.

Reply:
Yup, I realized this was an issue with this implementation -- we probably need to move it around somewhere. 

Aggregating the metric sounds interest as well, so as not to explode the cardinality?